wolfssl/ssl.h

```diff
@@ -3206,6 +3206,45 @@ WOLFSSL_API void  wolfSSL_CTX_SetRsaDecCb(WOLFSSL_CTX*, CallbackRsaDec);
 WOLFSSL_API void  wolfSSL_SetRsaDecCtx(WOLFSSL* ssl, void *ctx);
 WOLFSSL_API void* wolfSSL_GetRsaDecCtx(WOLFSSL* ssl);
 #endif
+
+/* Protocol Callback */
+typedef int (*CallbackGenMasterSecret)(WOLFSSL* ssl, void* ctx);
+WOLFSSL_API void  wolfSSL_CTX_SetGenMasterSecretCb(WOLFSSL_CTX*, CallbackGenMasterSecret);
+WOLFSSL_API void  wolfSSL_SetGenMasterSecretCtx(WOLFSSL* ssl, void *ctx);
+WOLFSSL_API void* wolfSSL_GetGenMasterSecretCtx(WOLFSSL* ssl);
+
+typedef int (*CallbackGenPreMaster)(WOLFSSL* ssl, byte *premaster,
+                                                   word32 preSz, void* ctx);
+WOLFSSL_API void  wolfSSL_CTX_SetGenPreMasterCb(WOLFSSL_CTX*, CallbackGenPreMaster);
+WOLFSSL_API void  wolfSSL_SetGenPreMasterCtx(WOLFSSL* ssl, void *ctx);
+WOLFSSL_API void* wolfSSL_GetGenPreMasterCtx(WOLFSSL* ssl);
+
+typedef int (*CallbackGenSessionKey)(WOLFSSL* ssl, void* ctx);
+WOLFSSL_API void  wolfSSL_CTX_SetGenSesssionKeyCb(WOLFSSL_CTX*, CallbackGenSessionKey);
+WOLFSSL_API void  wolfSSL_SetGenSesssionKeyCtx(WOLFSSL* ssl, void *ctx);
+WOLFSSL_API void* wolfSSL_GetGenSesssionKeyCtx(WOLFSSL* ssl);
+
+typedef int (*CallbackSetKeys)(WOLFSSL* ssl, void* ctx);
+WOLFSSL_API void  wolfSSL_CTX_SetSetKeysCb(WOLFSSL_CTX*, CallbackSetKeys);
+WOLFSSL_API void  wolfSSL_SetSetKeysCtx(WOLFSSL* ssl, void *ctx);
+WOLFSSL_API void* wolfSSL_GetSetKeysCtx(WOLFSSL* ssl);
+
+typedef int (*CallbackTlsFinished)(WOLFSSL* ssl,
+                            const byte *side, 
+                            const byte *handshake_hash,
+                            byte *hashes, void* ctx);
+WOLFSSL_API void  wolfSSL_CTX_SetTlsFinishedCb(WOLFSSL_CTX*, CallbackTlsFinished);
+WOLFSSL_API void  wolfSSL_SetTlsFinishedCtx(WOLFSSL* ssl, void *ctx);
+WOLFSSL_API void* wolfSSL_GetTlsFinishedCtx(WOLFSSL* ssl);
+
+#if !defined(WOLFSSL_NO_TLS12) && !defined(WOLFSSL_AEAD_ONLY)
+typedef int (*CallbackVerifymac)(WOLFSSL *ssl, const byte* message, 
+                    word32 messageSz, word32 macSz, word32 content, void* ctx);
+WOLFSSL_API void  wolfSSL_CTX_SetVerifymacCb(WOLFSSL_CTX*, CallbackVerifymac);
```

Comment:
As mentioned above, maintain camel case.  These APIs should have `Mac` capitalized. Example: `wolfSSL_CTX_SetVerifyMacCb`

Reply:
Changed to capitalized API name